lib/BridgedRoom.js

```diff
@@ -264,39 +265,80 @@ BridgedRoom.prototype._handleSlackMessage = function(message, ghost) {
     else if (message.files) {
         for (var i = 0; i < message.files.length; i++) {
             const file = message.files[i];
-            // We also need to upload the thumbnail
-            let thumbnail_promise = Promise.resolve();
-            // Slack ain't a believer in consistency.
-            const thumb_uri = file.thumb_video || file.thumb_360;
-            if (thumb_uri) {
-                thumbnail_promise = ghost.uploadContentFromURI(
-                    {
-                        // Yes, we hardcode jpeg. Slack always use em.
-                        title: `${file.name}_thumb.jpeg`,
-                        mimetype: "image/jpeg",
-                    },
-                    thumb_uri,
-                    this._slack_bot_token
-                );
+            if (file.mode === "snippet") {
+                var options = url.parse(file.url_private);
```

Comment:
Somewhere in the code I did this with ``rp`` and it was a lot less painful than directly using ``https``. You could save yourself a lot of code doing it :)

Reply:
So https://github.com/matrix-org/matrix-appservice-slack/blob/develop/lib/SlackGhost.js#L182 I download a file with ``rp`` which you should be able to steal for your purposes?